ceph_iscsi_config/gateway.py

```diff
@@ -402,13 +402,20 @@ def map_luns(self, config):
 
         lio_root = root.RTSRoot()
 
+        # a dict, key with tpg and storage object name
+        tpg_stg_object_dict = {}
+        for tpg in self.tpg_list:
+
+            for l in tpg.luns:
+                tpg_stg_object_dict[str(tpg.tag) + '-' + l.storage_object.name] = l.storage_object.name
+
         # process each storage object added to the gateway, and map to the tpg
         for stg_object in lio_root.storage_objects:
 
             for tpg in self.tpg_list:
                 self.logger.debug("processing tpg{}".format(tpg.tag))
 
-                if not self.lun_mapped(tpg, stg_object):
+                if not tpg_stg_object_dict.has_key(str(tpg.tag) + "-" + stg_object.name):
```

Comment:
I don't think we actually need to call lun_mapped for this as LUN() checks if the object is already mapped for us. We could just do:

```
for stg_object in lio_root.storage_objects:
    for tpg in self.tpg_list:
        try:
            mapped_lun = LUN(tpg, lun=lun_id, storage_object=stg_object)
            ....
        except RTSLibError as err:
            if "already exists in configFS" not in str(err):
                self.logger.error("LUN mapping failed: {}".format(err))
                self.error = True
                self.error_msg = str(err)
                return

           #Already created. Ignore and loop to the next tpg.
           continue
.....

```

Oh yeah, could your post your patch to this GH:

https://github.com/ceph/ceph-iscsi

instead of ceph-iscsi-config/cli?

ceph-iscsi is the repo that will be used going forward. ceph-iscsi-config/cli is more only used for a older version of the tools/lib.

Reply:
ok

Reply:
when catch a exception, configuration will be cleaned and the program exit.